tools/release-init.sh

```diff
@@ -34,8 +36,6 @@ git submodule update --init --recursive
 
 project_dir=$(git rev-parse --show-toplevel)
 
-git checkout -b release/automated_v$realm_version
```

Comment:
It appears that already being on the expected branch confuses the the make-release step [here](https://github.com/realm/realm-core/blob/master/.github/workflows/prepare-release.yml#L28) so this line was unnecessary.